packages/lesswrong/components/dialogues/CalendlyIFrame.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import { registerComponent } from '../../lib/vulcan-lib'
+import { calendlyPreviewStyles } from '../../themes/stylePiping'
+
+const styles = (theme: ThemeType) => ({
+  ...calendlyPreviewStyles(theme),
+  calendlyEmbed: {
+    ...calendlyPreviewStyles(theme)["& div.calendly-preview"]
+  }
+})
+
+export const validatedCalendlyUrl = (url: string) => {
+  if (url === "") return {valid: true, url: ""}
+  let calendlyUrl = null
+  try {
+    calendlyUrl = new URL(url.slice(0,4) === "http" ? url : `https://${url}`)
```

Comment:
what is `url.slice(0,4) === "http" ? url : `https://${url}`` doing?

Reply:
`new URL("www.calendly.com")` is a type error, so we want to add the protocol to the front if it's not there

Reply:
Oh, I see.  I think you want `.startsWith`, which is much clearer (semantically)